### PR TITLE
User/simore/master itsm service now controller

### PR DESCRIPTION
**Pull request by sidmore**
<!--- This repository only accepts pull requests related to open issues, please link the open issue in description below. See https://help.github.com/articles/closing-issues-using-keywords/ to learn about automation. 
ITSMSkill should be able to process Any Ticket Updates for example when Urgency is changed. This PR is an enabler to receive the incoming request from ServiceNow and create an Adaptive card of the change 

### Purpose
*What is the context of this pull request? Why is it being done?*
a. Implement ServiceNowController to process ServiceNow Post request
b. Implement ServiceNowMessageReceiver to create an EventActivity
c. Add ServiceNow Event to OnEventActivityCallback in DefaultActivityHandler
d. Add ServiceNowController Test

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*
ServiceNowUpdate2

### Tests
*Is this covered by existing tests or new ones? If no, why not?*
a. Added Controller Test
b. Need to figure out how to implement Notification test

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [x] I have commented my code, particularly in hard-to-understand areas	
- [x] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: skills/csharp/experimental/itsmskill/Bots/DefaultActivityHandler.cs [hunk 1]
```diff
@@ -53,9 +68,149 @@ protected override Task OnTeamsSigninVerifyStateAsync(ITurnContext<IInvokeActivi
...
+                                Columns = new List<AdaptiveColumn>
+                                {
+                                    new AdaptiveColumn
+                                    {
+                                        Width = AdaptiveColumnWidth.Stretch,
+                                        Items = new List<AdaptiveElement>
+                                        {
+                                            new AdaptiveTextBlock
+                                            {
+                                                Text = $"Title: {serviceNowNotification.Title}",
```
Comment (pavolum) [inline skills/csharp/experimental/itsmskill/Bots/DefaultActivityHandler.cs:128]: I assume ITSM is not intended to support multi language, if it is expected to work across languages then these will need to be localized.

File: skills/csharp/experimental/itsmskill/Bots/DefaultActivityHandler.cs [hunk 2]
```diff
@@ -53,9 +68,149 @@ protected override Task OnTeamsSigninVerifyStateAsync(ITurnContext<IInvokeActivi
...
+                        break;
+                    }
+
+                case ServiceNowEvents.Proactive:
+                    {
+                        var eventData = JsonConvert.DeserializeObject<ServiceNowNotification>(turnContext.Activity.Value.ToString());
+
+                        var proactiveModel = await _proactiveStateAccessor.GetAsync(turnContext, () => new ProactiveModel());
+
+                        // TODO: Implement a proactive subscription manager for mapping Notification to ConversationReference
```
Comment (pavolum) [inline skills/csharp/experimental/itsmskill/Bots/DefaultActivityHandler.cs:91]: Add a task for this in GH for tracking, if it is something that will need to be implemented after this PR

**Review by pavolum — APPROVED** [review]
Great PR, you really know your bot stuff :D 

Couple minor comments on my part...
- You will need to push new changes to Next, not to Master. Please start a new PR pushing from your branch to Next and let me know so I can re-approve there

- Most experimental skills are not multi lang support, if thats the case for this one then ignore my localization comment.

- Do we have any documentation on proactive messaging? If so you should add a link to this skill (once it is pushed) in the documentation for folks to have an E2E sample

- Looks like a broken build, if the build broke as a result to your code change then you may need to change tests/code.

File: skills/csharp/experimental/itsmskill/Bots/DefaultActivityHandler.cs (hunk 1 above)
Comment (ryanisgrig) [reply skills/csharp/experimental/itsmskill/Bots/DefaultActivityHandler.cs:128]: ITSM is in preview and thus English only

**Comment by ryanisgrig** [comment]
FYI you can change a PR's base above
image

I tried rebasing this to `next` but there are more files in that PR than the 14 changed here.

File: skills/csharp/experimental/itsmskill/Bots/DefaultActivityHandler.cs (hunk 2 above)
Comment (ryanisgrig) [reply skills/csharp/experimental/itsmskill/Bots/DefaultActivityHandler.cs:91]: By the way, there is an app available for this that can automatically create issues based on TODO comments: https://probot.github.io/apps/todo/

**Review by ryanisgrig — CHANGES_REQUESTED** [review]
This PR needs to be made against the [botframework-skills repository](https://github.com/microsoft/botframework-skills).

**Pull request closed without merging**